esmvalcore/preprocessor/_volume.py

```diff
@@ -176,16 +176,22 @@ def volume_statistics(
 
     The volume average is weighted acoording to the cell volume. Cell volume
     is calculated from iris's cartography tool multiplied by the cell
-    thickness.
+    thickness. If fx_files is provided and the variable object has
+    fx_var_preprocess
+    set to True, the variable's preprocessor chain will be applied to each of
+    the requested fx variables, up to and not including this step.
 
     Parameters
     ----------
         cube: iris.cube.Cube
             Input cube.
         operator: str
             The operation to apply to the cube, options are: 'mean'.
-        fx_files: dict
-            dictionary of field:filename for the fx_files
+        fx_files: list
+            list of field:short_name for the fx variables requested or
+            list of field:dict for the fx variables requested, including
+            but not limited to: short_name, mip, experiment etc (at least
+            short_name required)
```

Comment:
This is not the correct description for the `fx_files` variable used in this function, but for the `fx_files` the user has to specify in the recipe (this is changes in `_recipe.py`). Do we want to change that so that the function docstring corresponds to the actual function itself?

Reply:
Whats wrong with it? They list of strings or dicts

Reply:
https://github.com/ESMValGroup/ESMValCore/blob/2b46632fc0d1403cc7398f7d126fd71b9d598fe7/esmvalcore/preprocessor/_volume.py#L216

No, they need to be `dict`. This is converted in `recipe.py`

Reply:
Ah from your latest code changes, yesh

Reply:
I didn't change anything in the functions `area_statistics` and `volume_statistics` themselves :grin:

Reply:
Ok but then volume statistics and area statistics can simply take fx_files: ['volcello'] so they dont need to be only dicts

Reply:
Yes they do need to be `dict`s. Look at the function:

https://github.com/ESMValGroup/ESMValCore/blob/2b46632fc0d1403cc7398f7d126fd71b9d598fe7/esmvalcore/preprocessor/_volume.py#L170-L224

The option you describe in the docstring are the options in the recipe. The are changed here

https://github.com/ESMValGroup/ESMValCore/blob/2b46632fc0d1403cc7398f7d126fd71b9d598fe7/esmvalcore/_recipe.py#L508-L525

to be `dict`s eventually.

Reply:
Aye, am on the stoopid phone so Iit's hard to see the code so I trust you :grin: